### PR TITLE
Dependencies and GH workflow updated

### PR DESCRIPTION
**Pull request by ashtilawat23**
### What work was done? 
- Dependencies were updated to the latest versions.
- Unused dependencies were removed from starter code. 
- Some dependencies were moved to devDependencies.
- SonarCloud was removed from the GH workflow.
- Engine was updated in package to look for Node v14 or higher. 

### Why was this work done? 
- To update template node-api-starter repo.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -23,12 +23,20 @@ jobs:
           node-version: "12"
       - run: cp .env.sample .env
```
Comment (ryan-hamblin) [inline .github/workflows/ci.yml:24]: Is the only reason we need an `.env` file for this environment for tests that include "Okta"?

Comment (ryan-hamblin) [reply .github/workflows/ci.yml:24]: Like, do we need to run this `cp` command for an env vile to run a linter?

**Review by ryan-hamblin — CHANGES_REQUESTED** [review]
Everything here is great! I'm glad we're upgrading and maybe a bunch of these fancy "Dependabot" upgrades go away now. 

I just had one question about the necessity of the `.env` file for running lint.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ashtilawat23) [reply .github/workflows/ci.yml:24]: I think these are leftovers from when we used to use a SONAR_TOKEN and GITHUB_TOKEN. You can see the tokens in the previous workflow on main. I agree. I will remove them.

**Comment by ashtilawat23** [comment]
Screen Shot 2022-03-07 at 6 02 54 PM
@ryan-hamblin Looks like the line of code is necessary for this test to pass.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (tinomen) [reply .github/workflows/ci.yml:24]: The app can't run if the env variables don't exist. I think it's mostly an issue with knex. For now we just need to use the dummy sample .env file.

Comment (tinomen) [reply .github/workflows/ci.yml:24]: it's also the cheapest of all the run tasks in the job so doesn't "hurt" anything.

**Review by ryan-hamblin — APPROVED** [review]

**Pull request merged**